src/utils/parallelPaginatedRequest.js

```diff
@@ -0,0 +1,75 @@
+import axios from 'axios';
+
+async function fetchPage(apiUrl, requestConfig, page, pageSize, maxRetries) {
+  let attempt = 0;
+  let data = null;
+
+  while (attempt < maxRetries) {
+    attempt++;
+
+    try {
+      const response = await axios.get(`${apiUrl}&page=${page}&page_size=${pageSize}`, requestConfig);
+
+      if (response.status !== 200) {
+        throw new Error(`HTTP error ${response.status}`);
+      }
+
+      data = response.data.data;
+      break;
+    } catch (error) {
+      console.error(`Error fetching page ${page} (attempt ${attempt}):`, error.message);
+    }
+  }
+
+  if (data === null) { // just logging in error, this place can be used to trigger a callback on maxRetries exhausted per page
+    console.error(`Failed to fetch page ${page} after ${maxRetries} attempts.`);
```

Comment:
This is logging on fetch cancellations, which are intentional and happen quite frequently. As a result, the console is getting spammed in Chrome. Can we tighten that up to only log errors if they're not cancellations?